activiti-cloud-service-common/activiti-cloud-services-common-security/src/main/java/org/activiti/cloud/services/common/security/jwt/JwtUserInfoUriAuthenticationConverter.java

```diff
@@ -63,19 +68,30 @@ public void setUsernameClaim(String usernameClaim) {
     public String getPrincipalClaimName(Jwt jwt) {
         String username = jwt.getClaimAsString(usernameClaim);
         if (username == null) {
-            Instant issuedAt = jwt.getIssuedAt();
-            Instant expiresAt = jwt.getExpiresAt();
-            OAuth2AccessToken accessToken = new OAuth2AccessToken(
-                TokenType.BEARER,
-                jwt.getTokenValue(),
-                issuedAt,
-                expiresAt
-            );
-            OAuth2UserRequest userRequest = new OAuth2UserRequest(clientRegistration, accessToken);
             String cacheKey = getCacheKey(jwt);
             if (cacheKey != null) {
-                OAuth2User oAuth2User = this.oAuth2UserServiceCacheable.loadUser(userRequest, cacheKey);
-                username = oAuth2User.getName();
+                if (jwt.getClaimAsStringList("scope").contains("openid")) {
```

Comment:
add a null pointers check on jwt.getClaimAsStringList("scope")., 
```suggestion
                String scope = jwt.getClaimAsStringList("scope");
                if (scope != null && scope.contains("openid")) {
```